modules/library/main/src/main/java/org/geotools/renderer/crs/AzimulthalEquidistantProjectionHandlerFactory.java

```diff
@@ -161,12 +161,37 @@ private void initializeDatelineCutter(CoordinateReferenceSystem crs, Point2D.Dou
                 Geometry difference = renderingGeometry.difference(bufferedDateline);
                 List<Polygon> polygons = PolygonExtracter.getPolygons(difference);
                 for (Polygon p : polygons) {
-                    Geometry transformed =
-                            JTS.transform(
-                                    p,
-                                    CRS.findMathTransform(
-                                            renderingEnvelope.getCoordinateReferenceSystem(),
-                                            sourceCRS));
+                    Geometry transformed = null;
+                    try {
+                        transformed =
+                                JTS.transform(
+                                        p,
+                                        CRS.findMathTransform(
+                                                renderingEnvelope.getCoordinateReferenceSystem(),
+                                                sourceCRS));
+                    } catch (TransformException e) {
+                        // uh oh, got into a special case of the target projection... go step by
+                        // step
+                        Geometry polygonWGS84 =
+                                JTS.transform(
+                                        p,
+                                        CRS.findMathTransform(
+                                                renderingEnvelope.getCoordinateReferenceSystem(),
+                                                DefaultGeographicCRS.WGS84));
+                        ProjectionHandler handler =
+                                ProjectionHandlerFinder.getHandler(
+                                        new ReferencedEnvelope(sourceCRS),
+                                        DefaultGeographicCRS.WGS84,
+                                        false);
+                        if (handler == null) throw e;
+                        Geometry preProcessed = handler.preProcess(polygonWGS84);
+                        transformed =
+                                JTS.transform(
+                                        preProcessed,
+                                        CRS.findMathTransform(
+                                                renderingEnvelope.getCoordinateReferenceSystem(),
+                                                sourceCRS));
```

Comment:
I thought through putting a try catch around these, in order to throw the original TransformException exception. It is more useful to have this failure so you can tell exactly how far it got.

Reply:
The method declares to throw TransformException, so in case JTS.transform throws, it's going to go out of the method and handled at higher level, logged or re-wrapped. Don't think anything is lost. Have I misunderstood your comment?

Reply:
Nope it is fine, was trying to decide if there was any value in throwing the origional exception (there was not)